edx_proctoring/tests/test_utils.py

```diff
@@ -70,3 +75,35 @@ def test_emit_event(self):
                 'one': 'two'
             }
         )
+
+    @ddt.data(
+        *product(
+            [
+                ProctoredExamStudentAttemptStatus.started,
+                ProctoredExamStudentAttemptStatus.ready_to_submit,
+            ],
+            [
+                ProctoredExamStudentAttemptStatus.created,
+                ProctoredExamStudentAttemptStatus.download_software_clicked,
+                ProctoredExamStudentAttemptStatus.ready_to_start,
+            ]
+        )
+    )
+    @ddt.unpack
+    def test_is_reattempting_exam_from_in_progress_state(self, from_status, to_status):
+        """Tests that re-attempting exam returns true while transiting from given statuses"""
+        self.assertTrue(is_reattempting_exam(from_status, to_status))
+
+    @ddt.data(
+        ProctoredExamStudentAttemptStatus.created,
+        ProctoredExamStudentAttemptStatus.download_software_clicked,
+        ProctoredExamStudentAttemptStatus.ready_to_start,
+        ProctoredExamStudentAttemptStatus.ready_to_submit,
```

Comment:
I don't think that this test works correctly for the status `ProctoredExamStudentAttemptStatus. ready_to_submit `. My understanding of this test is that you you're testing the case 

```
...
    if not ProctoredExamStudentAttemptStatus.is_in_progress_status(from_status):
        return False
...
```

in the `is_reattempting_exam` util function. Is that right?

If so, `ProctoredExamStudentAttemptStatus.ready_to_submit` isn't going to hit that return statement. It's going to drop down to `return ProctoredExamStudentAttemptStatus.is_pre_started_status(to_status)`. And, obviously, `foo` isn't a valid status, so that will return `False`. 

If you're testing that, no matter what the `to_status` is, for the provided statues, you'll always get `False`, you should remove `ProctoredExamStudentAttemptStatus.ready_to_submit`.

Reply:
Perfect, thanks for the all the explanation, I got your point. 👍